teps/0027-https-connection-to-triggers-eventlistener.md

```diff
@@ -0,0 +1,135 @@
+
+### Goals
+
+* Make an additive change to the EventListener API to support `HTTPS` connection.
+
+### Non-Goals
+
+* Creating and maintaining certificates which are required for `HTTPS` connection.
+
+## Proposal
+
+Triggers support end to end secure `HTTPS` connection to eventlistener pod along with existing `HTTP` connections.
+
+### User Stories
+
+* User want to replace parts of their (Jenkins) build infrastructure with a cloud-native build infrastructure,
+and their security requirements specify that TLS needs to be used for both communication outside the cluster and 
+inside the cluster then having this solution would help. 
+
+* Users using custom/inbuilt interceptors like Github want their request to have secure connection all the way to the tekton trigger eventlistener pod then that can be solved with this feature.
+
+### Usage examples
+
+```yaml
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: EventListener
+metadata:
+  name: github-listener-interceptor
+spec:
+  triggers:
+    - name: github-listener
+      interceptors:
+        - github:
+            secretRef:
+              secretName: github-secret
+              secretKey: secretToken
+            eventTypes:
+              - pull_request
+      bindings:
+        - ref: github-pr-binding
+      template:
+        name: github-template
+  resources:
+    kubernetesResource:
+      spec:
+        template:
+          spec:
+            serviceAccountName: tekton-triggers-github-sa
+            volumes:
+              - name: https-connection
+                secret:
+                  secretName: ssh-key-secret
+            containers:
+            - volumeMounts:
+                - name: https-connection
+                  mountPath: "/etc/triggers/ssl/"
+                  readOnly: true
+              env:
+              - name: SSL_CERT_DIR
+                value: "/etc/triggers/ssl/tls.crt"
+              - name: SSL_KEY_DIR
+                value: "/etc/triggers/ssl/tls.key"
+```
+
+## Design Details
+
+The main goal of this TEP is to make triggers flexible enough to configure both `HTTPS` and `HTTP` connections with simple configuration changes to EventListener.
+
+With the help of `podtemplate` as part of `kubernetesResource` user mount their certificates using volumes
+
+ex:
+```yaml
+volumes:
+  - name: https-connection
+    secret:
+      secretName: ssh-key-secret
+containers:
+- volumeMounts:
+    - name: https-connection
+      mountPath: "/etc/triggers/ssl/"
+      readOnly: true
+```
+and provide location of mounted certs using Env.
+```yaml
+env:
+- name: SSL_CERT_DIR
+  value: "/etc/triggers/ssl/tls.crt"
+- name: SSL_KEY_DIR
+  value: "/etc/triggers/ssl/tls.key"
+```
+There are 2 env var reserved for `HTTPS` connection to read `key` and `cert`
+1. **SSL_CERT_DIR**: specify the location of mounted cert file.
+2. **SSL_KEY_DIR**: specify the location of mounted key file.
```

Comment:
This should suffixed by `_FILE` instead of `_DIR`.

Reply:
Changed `SSL_CERT_DIR`, `SSL_KEY_DIR` to `SSL_CERT_FILE`, `SSL_KEY_FILE`